commands/delete/endpoint/command.go

```diff
@@ -0,0 +1,175 @@
+// Package endpoint implements the 'delete endpoint' sub-command.
+package endpoint
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/fatih/color"
+	"github.com/giantswarm/gscliauth/config"
+	"github.com/giantswarm/microerror"
+	"github.com/spf13/cobra"
+
+	"github.com/giantswarm/gsctl/client"
+	"github.com/giantswarm/gsctl/commands/errors"
+	"github.com/giantswarm/gsctl/confirm"
+	"github.com/giantswarm/gsctl/flags"
+)
+
+// Arguments represents all argument that can be passed to our
+// business function.
+type Arguments struct {
+	// API endpoint to delete
+	APIEndpoint string
+	// Don't prompt
+	Force bool
+	// Verbosity
+	Verbose bool
+}
+
+func collectArguments(positionalArgs []string) Arguments {
+	var endpointToDelete string
+	if len(positionalArgs) > 0 {
+		endpointToDelete = positionalArgs[0]
+	}
+
+	return Arguments{
+		APIEndpoint: endpointToDelete,
+		Force:       flags.Force,
+		Verbose:     flags.Verbose,
+	}
+}
+
+var (
+	// Command performs the "delete endpoint" function
+	Command = &cobra.Command{
+		Use:   "endpoint",
+		Short: "Delete endpoint",
+		Long: `Deletes an API endpoint.
+
+Caution: This will remove the API endpoint from the existing configuration. To use it again, you will have to re-authenticate. There is no way to undo this.
+
+Example:
+
+	gsctl delete endpoint https://api.gigantic.io`,
+		PreRun: printValidation,
+		Run:    printResult,
+	}
+)
+
+func init() {
+	initFlags()
+}
+
+func initFlags() {
+	Command.ResetFlags()
+	Command.Flags().BoolVarP(&flags.Force, "force", "", false, "If set, no interactive confirmation will be required (risky!).")
+}
+
+// printValidation runs our pre-checks.
+// If errors occur, error info is printed to STDOUT/STDERR
+// and the program will exit with non-zero exit codes.
+func printValidation(cmd *cobra.Command, args []string) {
+	arguments := collectArguments(args)
+
+	err := validatePreconditions(arguments)
+	if err != nil {
+		client.HandleErrors(err)
+		errors.HandleCommonErrors(err)
+
+		var headline = ""
+		var subtext = ""
+
+		switch {
+		case errors.IsEndpointMissingError(err):
+			headline = "No API endpoint specified"
+			subtext = "See --help for usage details."
+		case errors.IsCouldNotDeleteEndpointError(err):
+			headline = "The API endpoint could not be deleted."
+			subtext = "You might try again in a few moments. If that doesn't work, please contact the Giant Swarm support team."
+			subtext += "Sorry for the inconvenience!"
+		default:
+			headline = err.Error()
+		}
```

Comment:
Here in `printValidation` we only have to handle the errors that can occur in `validatePreconditions`.

`errors.IsCouldNotDeleteEndpointError` is not returned by `validatePreconditions` so should not be handled here. Instead it has to be handled in the `printResult` function. Also `client.HandleErrors(err)` isn't needed (but also doesn't hurt) here, as there is no client usage in `validatePreconditions`.